### DISCUSSION TITLE
ifort on Windows  - any issue with libraries

### DISCUSSION THREAD

**Discussion by bitli** [fortran-lang/fpm#589] [General]
Hello,
I try to use ifort on Windows -10 to write an interface to portaudio.
Although I have the option `link = "portaudio_x64"`in the `[build]` section, the Link command does not include it. The compiler has the `-l` option (which I am not sure is correct for ifort) :
```
+ ifort  /warn:all /check:all /error-limit:1 /Od /Z7 /assume:byterecl /traceback /module:build\ifort_782887A201FCE169\ftn_portaudio /Ibuild\ifort_782887A201FCE169\ftn_portaudio  build\ifort_782887A201FCE169\ftn_portaudio\example_paex_sine.f90.o build\ifort_782887A201FCE169\ftn_portaudio\libftn_portaudio.a -lportaudio_x64 -o build\ifort_782887A201FCE169\example\paex_sine.exe
```

But the generated link command des not include that reference:
```
-out:build\ifort_782887A201FCE169\example\paex_sine.exe
-debug
-pdb:build\ifort_782887A201FCE169\example\paex_sine.pdb
-subsystem:console
-incremental:no
build\ifort_782887A201FCE169\ftn_portaudio\example_paex_sine.f90.o
build\ifort_782887A201FCE169\ftn_portaudio\libftn_portaudio.a
```

even if the path (`LIB`, `LIBPATH` ...) are set correctly. Manually typing the link commands adding the library reference (the path to `portaudi_x64.lib`) works well.

Is there something I did not understand or is this a bug to report (I hope this is the right forum to ask this question).
Thanks

**Comment by ghost** [comment]
It's a bug. Thanks for reporting, @bitli

**Reply by awvwgk** [reply]
Fixed with #590

**Comment by LKedward** [comment]
Thanks for reporting @bitli. Can you confirm whether the build process actually fails with a link error?
If not, does the executable run as you would expect?

*fpm* uses the compiler driver to invoke the linker, so the line you quoted with the link flag is actually the link command.

**Reply by bitli** [reply]
Thanks for following. I will report it as a bug in the github issues, with more information (may take day or so).

**Reply by bitli** [reply]
Issue created: https://github.com/fortran-lang/fpm/issues/591
Once I create the executable via the link command (adding the library reference), the executable runs as expected.
